### DISCUSSION TITLE
MVP by 13 Aug?

### DISCUSSION THREAD

**Discussion by jethrokuan** [org-roam/org-roam-ui#9] [Product]
Not to put any pressure, but I'm giving a short talk on Org-roam on 13 Aug and would also like to demo org-roam-ui during the talk. I think I can already show off the graph, but a couple more features (e.g. viewing) would make it look more polished. If uncomfortable, I can not demo it at all as well.

**Comment by tefkah** [comment]
That would be very cool! The main graph will be done by the end of this week at the latest (ideally in the next couple of days), so keep an eye out for that. We won't be implementing the file viewing immediately as we are not sure what kind of UX we want to go for there, but there is already an implementation of the viewer, I could spin up a separate branch for you that has that implemented, but I can't guarantee how well it will work by August 13th! It slides in a sidebar with an HTML render of the note. 
You can check the current readme for the planned features right now.

Which features would you like to see btw? For instance, how would you envision file viewing working?

**Reply by jethrokuan** [reply]
> You can check the current readme for the planned features right now.

👍 

> For instance, how would you envision file viewing working?

I'd imagine it would be a separate view (tab?) where it's mainly text, with the graph at the side. It could start with the entire list of nodes and the global graph. And clicking an any node would view the node, and show the local graph and backlinks in text in a sidebar. (i.e. like a web, read-only version of what's in Emacs)

**Reply by kirillrogovoy** [reply]
The reason we haven't been focusing on something like the node text viewer is mainly not wanting to re-implement anything that Emacs/orgmode/org-roam do well already. At least in the beginning. Focusing more on extending stuff rather than replacing stuff.
